### PR TITLE
Implemented Order Actions: ExternalPrint, Cancel, CreateInvoice and Warehouse Resourse specific fields and action

### PR DESCRIPTION
**Pull request by kwestground**
### Description
Please explain the changes you made here.

### Checklist
- [Y] Code compiles correctly
- [Y] Created tests which fail without the change (if possible)
- [Y] Extended the README / documentation, if necessary

### CODE REVIEW

File: Fortnox.NET/Services/OrderService.cs [hunk 1]
```diff
@@ -61,5 +61,36 @@ public static async Task<Order> CreateOrderAsync(FortnoxApiRequest request, Orde
...
             return (await FortnoxAPIClient.CallAsync(apiRequest)).Data;
         }
+
+        /// <summary>
+        /// This action is used to set the field Sent as true from an external system without generating a PDF.
+        /// </summary>
+        /// <param name="request">FortnoxApiRequest object</param>
+        /// <param name="orderNumber">Fortnox Order Number</param>
+        /// <returns></returns>
+        public static async Task<Order> PrintAsync(FortnoxApiRequest request, int orderNumber)
```
Comment (brokenprogrammer) [inline Fortnox.NET/Services/OrderService.cs:71]: The name of this method might be confused with the actual print action for orders. Perhaps a name like `ExternalPrintAsync` is more fitting?

File: Fortnox.NET/Services/OrderService.cs [hunk 2]
```diff
@@ -61,5 +61,36 @@ public static async Task<Order> CreateOrderAsync(FortnoxApiRequest request, Orde
 
             return (await FortnoxAPIClient.CallAsync(apiRequest)).Data;
         }
+
+        /// <summary>
```
Comment (brokenprogrammer) [inline Fortnox.NET/Services/OrderService.cs:65]: Its awesome that you've started adding documentation to the library! This is something that we've been wanting to get started on but didn't get to yet. 💯 👍

File: Fortnox.NET/Services/OrderService.cs [hunk 3]
```diff
@@ -61,5 +61,36 @@ public static async Task<Order> CreateOrderAsync(FortnoxApiRequest request, Orde
...
+        /// This action is used to set the field Sent as true from an external system without generating a PDF.
+        /// </summary>
+        /// <param name="request">FortnoxApiRequest object</param>
+        /// <param name="orderNumber">Fortnox Order Number</param>
+        /// <returns></returns>
```
Comment (brokenprogrammer) [inline Fortnox.NET/Services/OrderService.cs:70]: Perhaps fill in the returns as well.

File: Fortnox.NET/Services/OrderService.cs [hunk 4]
```diff
@@ -61,5 +61,36 @@ public static async Task<Order> CreateOrderAsync(FortnoxApiRequest request, Orde
...
+        public static async Task<Order> PrintAsync(FortnoxApiRequest request, int orderNumber)
+        {
+            var apiRequest =
+                new FortnoxApiClientRequest<SingleResource<Order>>(HttpMethod.Put, request.AccessToken, request.ClientSecret,
+                    $"{ApiEndpoints.Orders}/{orderNumber}/externalprint");
+            return (await FortnoxAPIClient.CallAsync(apiRequest)).Data;
+        }
+
+
+        public static async Task<Order> CancelAsync(FortnoxApiRequest request, int orderNumber)
```
Comment (brokenprogrammer) [inline Fortnox.NET/Services/OrderService.cs:80]: Would you like to add documentation for theese methods as well?

File: Fortnox.NET.Tests/Fortnox.NET.Tests/OrderTests.cs [hunk 5]
```diff
@@ -194,5 +193,85 @@ public void CreateOrderTest()
...
+                            {
+                                ArticleNumber = "1",
+                                OrderedQuantity = 1,
+                                AccountNumber = 3001
+                            }
+                        }
+                }).GetAwaiter().GetResult();
+            var createInvoiceResponse = OrderService.CreateInvoiceAsync(request, createOrderResponse.DocumentNumber).GetAwaiter().GetResult();
+
+            Assert.AreEqual("1", createInvoiceResponse.CustomerNumber);
```
Comment (brokenprogrammer) [inline Fortnox.NET.Tests/Fortnox.NET.Tests/OrderTests.cs:219]: The CreateInvoice action sets an `InvoiceReference` for the order in, a more interesting test would be to check if it actually got a new reference to an invoice.

**Review by brokenprogrammer — CHANGES_REQUESTED** [review]
Thank you once again for your contributions @kwestground! It is very much appreciated 🙏
I just had a few comments on the pull request, will merge once addressed!

**Review by brokenprogrammer — APPROVED** [review]
I've reviewed the pull request now and tested it and it looks great to me.
Once again thanks for your contributions @kwestground 🙏

**Comment by GGAlanSmithee** [comment]
Thanks for you contribution @kwestground 🥂

**Pull request merged**